internal/terraform/context_apply.go

```diff
@@ -168,11 +168,23 @@ func (c *Context) ApplyAndEval(plan *plans.Plan, config *configs.Config, opts *A
 		return nil, nil, diags
 	}
 
+	schemas, schemaDiags := c.Schemas(config, plan.PriorState)
+	diags = diags.Append(schemaDiags)
+	if diags.HasErrors() {
+		return nil, nil, diags
+	}
+
+	changes, err := plan.Changes.Decode(schemas)
+	if err != nil {
+		diags = diags.Append(err)
+		return nil, nil, diags
+	}
+
```

Comment:
We do this twice now, right after one another - on line 159 we're calling applyGraph in which it's doing exactly the same thing. Is it worth moving this up, and changing the args for applyGraph to accept the already decoded changes so we don't do that?

Reply:
Unfortunately combining those forces even more duplication elsewhere, however looking more closely at the graph builders, the transformers don't need decoded instance changes, so that part of the mechanical renaming can be mostly reverted. We then can just remove the extra Decode all.